pyccel/parser/semantic.py

```diff
@@ -3871,56 +3841,9 @@ def _visit_Assign(self, expr):
-            macro = self.scope.find(name, 'macros')
-            if macro is None:
-                rhs = self._visit(rhs)
-                if isinstance(rhs, (PythonMap, PythonZip, PythonEnumerate, PythonRange)):
-                    errors.report(f"{type(rhs)} cannot be saved to variables", symbol=expr, severity='fatal')
-            else:
-
-                # TODO check types from FunctionDef
-                master = macro.master
-                results = []
-                args = [self._visit(i) for i in rhs.args]
-                args_names = [arg.value.name for arg in args if isinstance(arg.value, Variable)]
-                d_m_args = {arg.value.name:arg.value for arg in macro.master_arguments
-                                  if isinstance(arg.value, Variable)}
-
-                lhs_iter = lhs
-
-                if not sympy_iterable(lhs_iter):
-                    lhs_iter = [lhs]
-                results_shapes = macro.get_results_shapes(args)
-                for m_result, shape, result in zip(macro.results, results_shapes, lhs_iter):
-                    if m_result in d_m_args and not result in args_names:
-                        d_result = self._infer_type(d_m_args[m_result])
-                        d_result['shape'] = shape
-                        tmp = self._assign_lhs_variable(result, d_result, None, new_expressions, False)
-                        results.append(tmp)
-                    elif result in args_names:
-                        _name = _get_name(result)
-                        tmp = self.get_variable(_name)
-                        results.append(tmp)
-                    else:
-                        # TODO: check for result in master_results
-                        errors.report(INVALID_MACRO_COMPOSITION, symbol=result,
-                            bounding_box=(self.current_ast_node.lineno, self.current_ast_node.col_offset),
-                            severity='error')
-
-                expr = macro.make_necessary_copies(args, results)
-                new_expressions += expr
-                args = macro.apply(args, results=results)
-                if isinstance(master.funcdef, FunctionDef):
-                    func_call = FunctionCall(master.funcdef, args, self.current_function_name)
-                    if new_expressions:
-                        return CodeBlock([*new_expressions, func_call])
-                    else:
-                        return func_call
-                else:
-                    # TODO treat interface case
-                    errors.report(PYCCEL_RESTRICTION_TODO,
-                                  bounding_box=(self.current_ast_node.lineno, self.current_ast_node.col_offset),
-                                  severity='fatal')
+            rhs = self._visit(rhs)
+            if isinstance(rhs, (PythonMap, PythonZip, PythonEnumerate, PythonRange)):
+                errors.report(f"{type(rhs)} cannot be saved to variables", symbol=expr, severity='fatal')
```

Comment:
This check looks very general to me, so why adding it only to the case where `rhs` is the result of a `FunctionCall`?

Reply:
Hmm, I didn't add anything in this PR so if it's showing up as new it is simply because the indentation has changed. But I would think that the listed objects can only be created by a `FunctionCall` hence why it is here.

Even if you do
```python
return range(10)
```
Internally we would understand that as
```python
result = range(10)
return result
```
And would still end up at this line

Reply:
Sorry, I really thought that you had added these lines in this PR!

Thanks for the explanation anyway!